tests/test_reader.py

```diff
@@ -423,6 +423,114 @@ def test_point():
         assert pt.mask == numpy.array([0])
         assert pt.band_names == ["b1"]
 
+        # buffer is useless with interpolate=False
+        with pytest.warns(UserWarning):
+            pt = reader.point(
+                src_dst,
+                [-57.566, 73.6885],
+                coord_crs="epsg:4326",
+                indexes=1,
+                nodata=1,
+                buffer=0.5,
+            )
+
+        # resampling_method is useless with interpolate=False
+        with pytest.warns(UserWarning):
+            pt = reader.point(
+                src_dst,
+                [-57.566, 73.6885],
+                coord_crs="epsg:4326",
+                indexes=1,
+                nodata=1,
+                resampling_method="bilinear",
+            )
+
+        # Interpolate=True but with Nearest, so no influence
+        pt = reader.point(
+            src_dst,
+            [-57.566, 73.6885],
+            coord_crs="epsg:4326",
+            indexes=1,
+            nodata=1,
+            interpolate=True,
+        )
+        assert pt.data == numpy.array([2800])
+        assert pt.mask == numpy.array([255])
+        assert pt.band_names == ["b1"]
```

Comment:
Does pt also keep track of array coordinates? e.g. `assert pt.xy = np.array([1090,1086])`
```
gdallocationinfo -r nearest tests/fixtures/cog.tif -wgs84 -57.566 73.68856
Report:
  Location: (1090P,1086L)
  Band 1:
    Value: 2800
```

Reply:
no but we could 👍